src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -805,20 +740,21 @@ int D4NFilterObject::D4NFilterReadOp::D4NFilterGetCB::handle_data(bufferlist& bl
     } else if (bl.length() == rgw_get_obj_max_req_size && bl_rem.length() == 0) { // if bl is the same size as rgw_get_obj_max_req_size, write it to cache
       std::string oid = this->oid + "_" + std::to_string(ofs) + "_" + std::to_string(bl_len);
       ofs += bl_len;
-      block.blockId = oid;
-      block.hostsList.push_back(blockDir->get_addr().host + ":" + std::to_string(blockDir->get_addr().port)); //check if this is the correct way to get the local addr
-      block.size = source->get_obj_size();
-      block.cacheObj.bucketName = source->get_bucket()->get_name();
+      block.version = oid;
```

Comment:
why is the version being set to the oid here? Didnt we decide to start with an empty string first?

Reply:
What should the blockID be in this case?

Reply:
The oid, we decided that we will set the initial version to an empty string.

Reply:
The blockID is type `uint64_t` while oid is type `string`. I don't think we want any casting here

Reply:
I spent some time thinking about this - let the blockId be uint64_t and set it offset in handle_data() method. The object name and bucket name are already a part of CacheObject.

Reply:
So, what should represent the block name in the policy's `entries_map`?